netmaster/objApi/apiController.go

```diff
@@ -120,15 +131,131 @@ func (ac *APIController) AppDelete(app *contivModel.App) error {
 	return nil
 }
 
+// FIXME: hack to allocate unique endpoint group ids
+var globalEpgID = 1
+
 // EndpointGroupCreate creates end point group
 func (ac *APIController) EndpointGroupCreate(endpointGroup *contivModel.EndpointGroup) error {
 	log.Infof("Received EndpointGroupCreate: %+v", endpointGroup)
+
+	// assign unique endpoint group ids
+	endpointGroup.EndpointGroupID = globalEpgID
+	globalEpgID = globalEpgID + 1
+
+	// Find the tenant
+	tenant := contivModel.FindTenant(endpointGroup.TenantName)
+	if tenant == nil {
+		return errors.New("Tenant not found")
```

Comment:
As i mentioned above. I'm not sure centralized error codes are a good idea. Lets discuss it offline.